### PR TITLE
Patch prop types missing.

### LINKED ISSUE #3969: [v3] SimpleForm / TabbedForm are missing PropTypes
SimpleForm and TabbedForm Components don't have propTypes defined.
SimpleFormView and TabbedFormView have them instead and is killing my intellisense ;-).
I'll add them myself, but i'm not sure which ones to include.

### PR DESCRIPTION
**Pull request by WiXSL**
Closes #3969

### CODE REVIEW

File: packages/ra-ui-materialui/src/form/SimpleForm.js [hunk 1]
```diff
@@ -63,6 +63,27 @@ const SimpleForm = ({ initialValues, defaultValue, saving, ...props }) => {
     );
 };
 
+SimpleForm.propTypes = {
+    basePath: PropTypes.string,
+    children: PropTypes.node,
+    className: PropTypes.string,
+    defaultValue: PropTypes.oneOfType([PropTypes.object, PropTypes.func]), // @deprecated
+    initialValues: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+    handleSubmit: PropTypes.func, // passed by react-final-form
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/form/SimpleForm.js]: no, SimpleForm *creates* a react-final-form form, so `handleSubmit` is never a prop of `SimpleForm`

File: packages/ra-ui-materialui/src/form/SimpleForm.js [hunk 2]
```diff
@@ -63,6 +63,27 @@ const SimpleForm = ({ initialValues, defaultValue, saving, ...props }) => {
...
+    invalid: PropTypes.bool,
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/form/SimpleForm.js]: `invalid` and `pristine` are react-final-form properties that come into existence *inside* `SiumpleForm`, so they should not be PropTypes

File: packages/ra-ui-materialui/src/form/SimpleForm.js [hunk 3]
```diff
@@ -63,6 +63,27 @@ const SimpleForm = ({ initialValues, defaultValue, saving, ...props }) => {
...
+    resource: PropTypes.string,
+    redirect: PropTypes.oneOfType([
+        PropTypes.string,
+        PropTypes.bool,
+        PropTypes.func,
+    ]),
+    submitOnEnter: PropTypes.bool,
+    toolbar: PropTypes.element,
+    undoable: PropTypes.bool,
+    validate: PropTypes.func,
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/form/SimpleForm.js]: missing `record`, `save`, `version`. Besides, any props passed to react-final-form's `Form` can be passed to `SimpleForm`, too.

File: packages/ra-ui-materialui/src/form/TabbedForm.js [hunk 4]
```diff
@@ -78,6 +78,34 @@ const TabbedForm = ({ initialValues, defaultValue, saving, ...props }) => {
     );
 };
 
+TabbedForm.propTypes = {
+    basePath: PropTypes.string,
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/form/TabbedForm.js]: same remarks

File: packages/ra-ui-materialui/src/list/DatagridLoading.js [hunk 5]
```diff
@@ -106,4 +107,14 @@ const DatagridLoading = ({
...
 );
 
+DatagridLoading.propTypes = {
+    classes: PropTypes.object,
+    className: PropTypes.string,
+    expand: PropTypes.oneOfType([PropTypes.element, PropTypes.elementType]),
+    hasBulkActions: PropTypes.bool,
+    nbChildren: PropTypes.number,
+    nbFakeLines: PropTypes.number,
+    size: PropTypes.string,
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/list/DatagridLoading.js]: only 'small' and 'medium' are valid values for this MUI option

Comment (WiXSL) [reply packages/ra-ui-materialui/src/list/DatagridLoading.js]: Done

File: packages/ra-ui-materialui/src/form/TabbedForm.js (hunk 4 above)
Comment (WiXSL) [reply packages/ra-ui-materialui/src/form/TabbedForm.js]: Done.

File: packages/ra-ui-materialui/src/form/SimpleForm.js (hunk 2 above)
Comment (WiXSL) [reply packages/ra-ui-materialui/src/form/SimpleForm.js]: Done.

File: packages/ra-ui-materialui/src/form/SimpleForm.js (hunk 1 above)
Comment (WiXSL) [reply packages/ra-ui-materialui/src/form/SimpleForm.js]: Done.

File: packages/ra-ui-materialui/src/form/SimpleForm.js (hunk 2 above)
Comment (WiXSL) [reply packages/ra-ui-materialui/src/form/SimpleForm.js]: Corrected.

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**